aws/src/integration/README.md

```diff
@@ -0,0 +1,197 @@
+export AWS_ACCESS_KEY_ID=ABCDEFG1234567890
+export AWS_SECRET_ACCESS_KEY=qwertyuiopasdfghjklzxcvbnm
+export AWS_REGION=us-east-1
+export AWS_CROSS_REGION=us-west-2
+export AWS_TEST_ACCOUNT_ID=123456789012
+export AWS_TEST_BUCKET=my-bucket
+export AWS_TEST_CROSS_REGION_BUCKET=my-cross-region-bucket
+
+# run all tests
+./gradlew :iceberg-aws:integrationTest
+
+# run a single test suite
+./gradlew :iceberg-aws:integrationTest --tests "org.apache.iceberg.aws.s3.TestS3FileIOIntegration"
+
+# run a single test
+./gradlew :iceberg-aws:integrationTest --tests "org.apache.iceberg.aws.s3.TestS3FileIOIntegration.testServerSideKmsEncryption"
+```
+
+## Permissions
+
+The following JSON payload describes the minimum set of permissions required to execute tests.
+Developers interested in running the test suite against a personal AWS account can use this for reference.
+The AWS Tests CI workflow uses the same set of permissions.
+To add new permissions in CI, please contact [dev@iceberg.apache.org](mailto:dev@iceberg.apache.org).
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "S3BucketPermissions",
+      "Effect": "Allow",
+      "Action": [
+        "s3:GetAccessPoint",
+        "s3:CreateAccessPoint",
+        "s3:DeleteAccessPoint",
+        "s3:ListAccessPoints",
+        "s3:ListBucket"
+      ],
+      "Resource": [
+        "arn:aws:s3:::iceberg-aws-ci-220427471354-us-east-1",
+        "arn:aws:s3:::iceberg-aws-ci-220427471354-us-west-2"
+      ]
+    },
+    {
+      "Sid": "S3ObjectPermissions",
+      "Effect": "Allow",
+      "Action": [
+        "s3:PutObject",
+        "s3:GetObjectAcl",
+        "s3:GetObject",
+        "s3:DeleteObject",
+        "s3:AbortMultipartUpload"
+      ],
+      "Resource": [
+        "arn:aws:s3:::iceberg-aws-ci-220427471354-us-east-1/*",
+        "arn:aws:s3:::iceberg-aws-ci-220427471354-us-west-2/*"
+      ]
+    },
+    {
+      "Sid": "GluePermissions",
+      "Effect": "Allow",
+      "Action": [
+        "glue:UpdateDatabase",
+        "glue:DeleteDatabase",
+        "glue:CreateTable",
+        "glue:GetTables",
+        "glue:GetTableVersions",
+        "glue:UpdateTable",
+        "glue:DeleteTable",
+        "glue:GetDatabases",
+        "glue:GetTable",
+        "glue:GetDatabase",
+        "glue:GetTableVersion",
+        "glue:CreateDatabase"
+      ],
+      "Resource": [
+        "arn:aws:glue:us-east-1:220427471354:catalog",
+        "arn:aws:glue:us-east-1:220427471354:database/iceberg_aws_ci_*",
+        "arn:aws:glue:us-east-1:220427471354:table/iceberg_aws_ci_*/*",
+        "arn:aws:glue:us-east-1:220427471354:userDefinedFunction/iceberg_aws_ci_*/*"
+      ]
+    },
+    {
+      "Sid": "STSPermissions",
+      "Effect": "Allow",
+      "Action": [
+        "sts:TagSession",
+        "sts:AssumeRole"
+      ],
+      "Resource": [
+        "arn:aws:iam::220427471354:role/iceberg-aws-ci-*"
+      ]
+    },
+    {
+      "Sid": "KMSKeyCreationPermissions",
```

Comment:
It is also unsafe to have KMS key creation permission exposed. If credential is exposed, a caller can start to create keys and use them to perform any operation. Similar to the assume role case, we should pre-create a KMS key to be used across tests.